server/src/storage/time_series.rs

```diff
@@ -74,13 +75,20 @@ impl CompressedTimeSeries {
         let complete_folder_path = folder_path.join("compressed");
         fs::create_dir_all(complete_folder_path.as_path())?;
 
-        // Create a path that uses the first timestamp as the file name to better support
-        // pruning data that is too new or too old when executing a specific query.
+        // Create a path that uses the first start timestamp and the last end timestamp as the file
+        // name to better support pruning data that is too new or too old when executing a specific query.
         let start_times: &TimestampArray = batch.column(2).as_any().downcast_ref().unwrap();
-        let file_name = format!("{}.parquet", start_times.value(0));
-        let file_path = complete_folder_path.join(file_name);
+        let end_times: &TimestampArray = batch.column(3).as_any().downcast_ref().unwrap();
 
-        StorageEngine::write_batch_to_apache_parquet_file(batch, file_path.as_path());
+        let file_name = format!(
+            "{}-{}.parquet",
+            start_times.value(0),
```

Comment:
Maybe add a doc comment to `append_segment()` that states that `segment: RecordBatch` is assumed to be sorted by time.

Reply:
Fixed.